docs/firestore-schema.md

```diff
@@ -0,0 +1,89 @@
+# Firestore database structure
+
+## Top level collections
+
+These are similar to Firebase.
+
+`authed, demo, dev, qa, tests, users`
+
+Within `authed`, there are several portal site names:
+
+- `learn_concord_org` (production)
+- `learn_portal_staging_concord_org`
+- `learn_staging_concord_org`
+- `learn-migrate_concord_org` (not sure of status, has only `documents`)
+
+Within `demo`, the names of demo spaces (eg, "CLUE")
+
+Within `dev`, UUIDs of dev instances.
+
+Within `qa`, UUIDs of test instances.
+
+`tests` is something different, doc TODO.
+
+`users` top level collection looks like a mistake.
+
+## Second level
+
+Collections within `(authed|demo|dev|qa)/{id}`:
+
+- classes
+- curriculum
+- documents
+- images
+- mcimages
+- mcsupports
+- offerings
+- users
+
+## Third level
+
+### Contents of `classes/{classId}`
+
+Fields:
+
+- id  (string, eg "60349")
+- name (string)
+- context_id (string, uuid)
+- network: (string, name of network)
+- teacher: (string, full name of teacher who created it)
+- uri: (uri on the portal)
+- teachers: (array of IDs of teachers) _this needs updating_
+
+### Contents of `curriculum/{docPath}`
+
+TODO
+
+### Contents of `documents/{docId}`
+
+Fields:
+
+- key: (string, a mobx id)
+- title: (string)
+- type: (string, eg "problem")
+- uid: (string, id of user who owns this document)
```

Comment:
I think this is sometimes the owner of the real document, and sometimes it is the owner of the comments located under the metadata document. I'm basing that on the query used in `commented-documents.tsx`. But to know for sure we need to look at the firestore function that creates the documents and look at how it is called.